WebApp/autoreduce_webapp/reduction_viewer/models.py

```diff
@@ -20,34 +21,42 @@ class Status(models.Model):
 
     def __unicode__(self):
         return u'%s' % self.value        
-        
+
+
 class ReductionRun(models.Model):
-    run_number = models.IntegerField(blank=False)
-    run_version = models.IntegerField(blank=False)
-    run_name = models.CharField(max_length=200, blank=True)
-    experiment = models.ForeignKey(Experiment, blank=False, related_name='reduction_runs')
-    instrument = models.ForeignKey(Instrument, related_name='reduction_runs', null=True)
-    
-    script = models.TextField(blank=False)
-    
-    status = models.ForeignKey(Status, blank=False, related_name='+')
-    created = models.DateTimeField(auto_now_add=True, blank=False)
-    last_updated = models.DateTimeField(auto_now=True, blank=False)
-    started = models.DateTimeField(null=True, blank=True)
-    finished = models.DateTimeField(null=True, blank=True)
+    # Integer fields
+    run_number = models.IntegerField(blank=False, validators=[MinValueValidator(0)])
+    run_version = models.IntegerField(blank=False, validators=[MinValueValidator(0)])
     started_by = models.IntegerField(null=True, blank=True)
+
+    # Char fields
+    run_name = models.CharField(max_length=200, blank=True)
+
+    # Text fields
+    admin_log = models.TextField(blank=True)
     graph = models.TextField(null=True, blank=True)
-    
     message = models.TextField(blank=True)
     reduction_log = models.TextField(blank=True)
-    admin_log = models.TextField(blank=True)
-    
-    retry_run = models.ForeignKey('self', on_delete=models.SET_NULL, null=True, blank=True)
+    # Scripts should be 10,000 chars or less. The DB supports up to 4GB strings here
+    script = models.TextField(blank=False, validators=[MaxLengthValidator(10000)])
```

Comment:
Just a quick thing I noticed, currently the WISH script is 40,000 chars so maybe this should be more like 100,000?

Reply:
Okay, I'm happy to raise the limit as required. I just don't want a 2b character entry to be accidentally written.

I'll put that change in now